sympy/matrices/expressions/matexpr.py

```diff
@@ -313,8 +314,8 @@ def __getitem__(self, key):
         raise IndexError("Invalid index, wanted %s[i,j]" % self)
 
     def _is_shape_symbolic(self) -> bool:
-        return (not isinstance(self.rows, (SYMPY_INTS, Integer))
-            or not isinstance(self.cols, (SYMPY_INTS, Integer)))
+        return (not isinstance(self.rows, (SYMPY_INTS, Integer, numbers.Integral))
```

Comment:
Is this needed? I don't think `self.rows` and `self.cols` should be anything but a SymPy object.

Reply:
I have never used sage, but it might be possible to trace down how something other than `int`/`Integer` ends up here by adding these lines to this method
```python
        if not isinstance(self.rows, (SYMPY_INTS, Integer)):
            print(type(self.rows), self)
        if not isinstance(self.cols, (SYMPY_INTS, Integer)):
            print(type(self.cols), self)
```
and running tests.